src/sandbox/mod.rs

```diff
@@ -0,0 +1,167 @@
+//! Basic Pod Sandbox types
+
+pub mod pinned;
+
+use crate::criapi::{PodSandboxConfig, PodSandboxMetadata};
+use anyhow::{Context, Result};
+use getset::Getters;
+use std::fmt;
+use uuid::Uuid;
+
+#[derive(Getters)]
+pub struct Sandbox<T> {
+    #[get = "pub"]
+    /// The unique identifier.
+    id: Uuid,
+
+    #[get = "pub"]
+    /// Full name of the sandbox.
+    name: String,
+
+    #[get = "pub"]
+    /// Pod sandbox configuration, usually passed via the Container Runtime Interface.
+    metadata: PodSandboxMetadata,
+
+    /// Trait implementation for creating the sandbox.
+    implementation: T,
+}
+
+pub trait Pod {
+    /// Run a previously created sandbox.
+    fn run(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    /// Stop a previously started sandbox.
+    fn stop(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    /// Remove a stopped sandbox.
+    fn remove(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl<T> Sandbox<T>
+where
+    T: 'static + Default + Pod,
+{
+    /// Create a new Sandbox form the provided config and implementation.
+    pub fn from(mut config: PodSandboxConfig) -> Result<Self> {
```

Comment:
Let's decouple from the CRI and pass what we need extracted from the config.

Reply:
Sounds good. Fixed that! :+1:

Reply:
it may behoove us to have an intermediate layer between server and sandbox constructors. In cri-o, server package is huge because it does a lot of the translation between cri to oci and various other structs. we could have a sandbox factory that takes the server config and the cri config and returns a created sandbox. there, we could branch on type and do the necessary conversions